test/unit/metrics/test_metricframe_bootstrap.py

```diff
@@ -576,3 +576,23 @@ def test_ci_properties_raise_on_uninitialized_bootstrap_params(
 
     with expectation:
         getattr(mf, method)
+
+
+def test_bootstrapping_can_handle_missing_sensitive_feature_values_in_samples() -> None:
+    y_true = [0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0]
+    y_pred = [0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0]
+    sensitive_features = ["a", "b", "c", "a", "b", "c", "a", "b", "c", "a", "b"]
+
+    # using this random state yields a sample with a missing value, "b", for the sensitive feature
+    random_state = np.random.RandomState(1234)
+
+    with does_not_raise():
+        MetricFrame(
+            metrics=count,
+            y_true=y_true,
+            y_pred=y_pred,
+            sensitive_features=sensitive_features,
+            ci_quantiles=[0.1, 0.5, 0.9],
+            n_boot=3,
+            random_state=random_state,
+        )
```

Comment:
Can we check anything else? I'm sort of concerned that if the PRNG changes on us, then this will work because the above seed _won't_ produce a sample with a missing 'b' value.

Reply:
Or only have a single 'b' in the sensitive features, which should make it much easier to hit the case of 'b' being absent

Reply:
Done @riedgar-ms :pray: I double checked that the new input and random state do yield a bootstrapped sample that is missing "b".